### PR TITLE
TS-4056: remove alloc_cache from NetAccept

### PR DESCRIPTION
**Pull request by oknet**
remove alloc_cache from NetAccept & net_accept() to avoid mem leak on ~NetAccept().
add a condition check for vc==NULL follow allocate_vc().

### CODE REVIEW

File: iocore/net/UnixNetAccept.cc [hunk 1]
```diff
@@ -112,9 +106,17 @@ net_accept(NetAccept *na, void *ep, bool blockable)
...
+      con.close();
+      goto Ldone;
+    }
+    
     count++;
-    na->alloc_cache = NULL;
-
+    NET_SUM_GLOBAL_DYN_STAT(net_connections_currently_open_stat, 1);
+    vc->id = net_next_connection_number();
+    vc->con = con;
```
Comment (SolidWallOfCode) [inline iocore/net/UnixNetAccept.cc]: I think `vc->con.move(con)` would be better here.

**Comment by SolidWallOfCode** [comment]
A few questions -

1. Why is it the destruction of a `NetAccept` a concern - I thought that only happened during process shutdown. Is there a case where it happens during process execution? Or this a general cleanup?

2. Is `net_accept` actually called? I think it might be called if a plugin calls `TSNetAccept` but I would be interested in confirmation of that.

3. Why not have the destructor of `NetAccept`

**Comment by oknet** [comment]
Answers：
1. to implement an one shot accept (like a ftp pasv mode)
2. accept_fn = net_accept, accept_fn is called from NetAccept
3. the alloc_cache point to a vc, the vc may be a UnixNetVConnection, sslNetVConnection, or others ...
should dealloc it by netAllocate.free(), sslNetAllocate.free() ...
It is hardly to get the vc's type inside the destructor of NetAccept.

File: iocore/net/UnixNetAccept.cc (hunk 1 above)
Comment (oknet) [inline iocore/net/UnixNetAccept.cc]: Connection::move() not existed, I'm set the con.fd to NO_FD to avoid con.fd closed by ~Connection()

**Pull request closed without merging**